Assets/HotmaxEditorVR/universalTransform.cs

```diff
@@ -0,0 +1,225 @@
+    private float scrollSpeed = .2f;
+    public float scrollDistance = 0;
+
+    private bool _selectedObjectIsActive = stateManager.selectedObjectIsActive;
+    private bool _rotationGizmoIsSelected = stateManager.rotationGizmoIsSelected;
+
+    private stateManager.rotationModes _rotationMode = stateManager.rotationMode;
+    private stateManager.editorModes _editorMode = stateManager.editorMode;
+
+    private float initialXControllerPosition;
+    private float currentXControllerPosition;
+
+    private float initialYControllerPosition;
+    private float currentYControllerPosition;
+
+    private List<Transform> transforms = new List<Transform>();
+
+    private void Awake()
+    {
+
+    }
+
+    private void OnEnable()
+    {
+        stateManager.selectedObjectIsActiveEvent += updateSelectedObjectIsActive;
+        stateManager.rotationGizmoIsSelectedEvent += updateRotationGizmoIsSelected;
+        stateManager.rotationModeEvent += updateRotationModeEvent;
+        stateManager.editorModeEvent += updateEditorMode;
+    }
+
+    protected virtual void OnDisable()
+    {
+        stateManager.selectedObjectIsActiveEvent -= updateSelectedObjectIsActive;
+        stateManager.rotationGizmoIsSelectedEvent -= updateRotationGizmoIsSelected;
+        stateManager.rotationModeEvent -= updateRotationModeEvent;
+        stateManager.editorModeEvent -= updateEditorMode;
+    }
+
+    void updateSelectedObjectIsActive(bool value)
+    {
+        _selectedObjectIsActive = value;
+
+        //this gets the initial offset of the object to the controller
+        //the purpose being to start the object off at the same place it was in before selecting it
+        _initialPadYPosition = inputManager.device2.GetAxis().y;
+        distToController = Vector3.Distance(transform.position, inputManager.hand2.transform.position);
+        scrollDistance = 0;
+    }
+
+
+    void updateRotationGizmoIsSelected(bool value)
+    {
+        _rotationGizmoIsSelected = value;
+        initialXControllerPosition = inputManager.hand2.transform.position.x;
+        initialYControllerPosition = inputManager.hand2.transform.position.y;
+        setAllRotationGizmosVisible(true);
+    }
+
+    void updateRotationModeEvent(stateManager.rotationModes value)
+    {
+        _rotationMode = value;
+        if (_rotationGizmoIsSelected)
+        {
+            if (_rotationMode == stateManager.rotationModes.xRotationMode)
+            {
+                setOneRotationGizmoVisible(0);
+            }
+            if (_rotationMode == stateManager.rotationModes.yRotationMode)
+            {
+                setOneRotationGizmoVisible(1);
+            }
+            if (_rotationMode == stateManager.rotationModes.zRotationMode)
+            {
+                setOneRotationGizmoVisible(2);
+            }
+        }
+    }
+
+    void updateEditorMode(stateManager.editorModes value)
+    {
+        _editorMode = value;
+    }
+
+    void setOneRotationGizmoVisible(int gizmo)
```

Comment:
condense setOneRotationGizmoVisible and setAllRotationsGizmosVisible to one function, that takes an enum argument - with NONE, X, Y, Z -- and switch statement with functionality

Reply:
Got it!